docs/spec.md

```diff
@@ -3003,14 +3003,21 @@ More formally, `result[result_index] = operand[operand_index]` where:
     `index_vector_dim` < `rank(start_indices)`.
   * `[start_indices[batch_index]]` otherwise.
 * For `d_operand` in `axes(operand)`,
-  * `full_start_index[d_operand] = clamp(start_index[d_start], 0, dim(operand, d_operand) - slice_sizes[d_operand])`
+  * `full_start_index[d_operand] = clamp(start_index[d_start], 0,
+    dim(operand, d_operand) - slice_sizes[d_operand])`
     if `d_operand = start_index_map[d_start]`.
   * `full_start_index[d_operand] = 0` otherwise.
+* For `d_operand` in `axes(operand)`,
+  * `full_batching_index[d_operand] =
+    batch_index[d_start - (d_start < index_vector_dim ? 0 : 1)]`
```

Comment:
wdyt about this pseudo code? it's similar to the `full_start_index` but the `[d_start - (d_start < index_vector_dim ? 0 : 1)]` notation is a bit bizzare, is there a nicer way to express this?